src/components/constraint.js

```diff
@@ -57,24 +56,56 @@ module.exports = {
       return;
     }
 
+    /** The body frame differs from the element frame */
+    var pivot = new CANNON.Vec3(data.pivot.x, data.pivot.y, data.pivot.z);
+    var pivotTarget = new CANNON.Vec3(data.pivotTarget.x, data.pivotTarget.y, data.pivotTarget.z);
+    var axis = new CANNON.Vec3(data.axis.x, data.axis.y, data.axis.z);
+    var axisTarget = new CANNON.Vec3(data.axisTarget.x, data.axisTarget.y, data.axisTarget.z);
+
+    /* 
+     * This is actually the simplest way even with the superficial code duplication
+     * Cannon's constraint interface's inconsistencies makes it so.
+     */
     switch (data.type) {
+      case 'lock':
+        this.constraint = new CANNON.LockConstraint(this.el.body, data.target.body, { "maxForce": data.maxForce });
+        break;
       case 'distance':
-        this.constraint = new CANNON.DistanceConstraint(
-          el.body,
-          data.target.body,
-          options.distance || undefined,
-          options.maxForce
-        );
+        this.constraint = new CANNON.DistanceConstraint(this.el.body, data.target.body, data.distance, data.maxForce);
         break;
-      case 'lock':
-        this.constraint = new CANNON.LockConstraint(el.body, data.target.body, options);
+      case 'hinge':
+        this.constraint = new CANNON.HingeConstraint(
+          this.el.body,
+          data.target.body, {
+            "pivotA": pivot,
+            "pivotB": pivotTarget,
+            "axisA": axis,
+            "axisB": axisTarget,
+            "maxForce": data.maxForce
+          });
         break;
       case 'coneTwist':
-      case 'hinge':
+        this.constraint = new CANNON.ConeTwistConstraint(
+          this.el.body,
+          data.target.body, {
+            "pivotA": pivot,
+            "pivotB": pivotTarget,
+            "axisA": axis,
+            "axisB": axisTarget,
+            "maxForce": data.maxForce
+          });
+        break;
       case 'pointToPoint':
+        this.constraint = new CANNON.PointToPointConstraint(
+          this.el.body,
+          pivot,
+          data.target.body,
+          pivotTarget,
+          data.maxForce);
+        break;
+      default:
         throw new Error('[constraint] Unimplemented type.');
```

Comment:
Since it looks like you've implemented all the constraint types, could you change this message to: 

```
'[constraint] Unexpected type: ' + data.type
```